### PR TITLE
feat(Toggle): add Toggle component

### PR DESCRIPTION
**Pull request by shubhnik**

### CODE REVIEW

File: src/atoms/Toggle/Toggle.native.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import React, { useState, useContext, useRef } from 'react';
+import { View, Animated, TouchableOpacity } from 'react-native';
```
Comment (dhruvdutt) [inline src/atoms/Toggle/Toggle.native.js]: Use our own `<View />` component.

File: src/atoms/Toggle/Toggle.native.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+import styled, { ThemeContext } from 'styled-components/native';
+import PropTypes from 'prop-types';
+
+import automationAttributes from '../../_helpers/automation-attributes';
+
+const styles = {
+  size({ size, theme }) {
+    switch (size) {
+      case 'medium':
```
Comment (dhruvdutt) [inline src/atoms/Toggle/Toggle.native.js]: Is there only one variation of this component?

File: src/atoms/Toggle/Toggle.native.js (hunk 1 above)
Comment (shubhnik) [reply src/atoms/Toggle/Toggle.native.js]: Rebased master, done 👍

File: src/atoms/Toggle/Toggle.native.js (hunk 2 above)
Comment (shubhnik) [reply src/atoms/Toggle/Toggle.native.js]: Yeah, currently there is a medium-size Toggle only, here to accommodate new sizes in the future.

File: src/atoms/Toggle/Toggle.stories.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { View } from 'react-native';
+import Toggle from './Toggle';
+
+storiesOf('Toggle', module)
+  .addParameters({
+    component: Toggle,
+  })
+  .add('Toggle', () => (
```
Comment (deorukhkarchaitanya) [inline src/atoms/Toggle/Toggle.stories.js]:
```suggestion
  .add('default', () => (
```

File: src/atoms/Toggle/Toggle.native.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      onValueChange(!toggleState);
+    });
+  };
+
+  const containerFlexDirection = toggleState ? 'row-reverse' : 'row';
+  const hitSlopWidth =
+    parseInt(containerSize.width, 10) -
+    parseInt(containerPadding, 10) -
+    parseInt(knobSize.width, 10);
+  const hitSlop = toggleState ? { left: hitSlopWidth } : { right: hitSlopWidth };
```
[the remark this answers is not in the document]
Comment (shubhnik) [reply src/atoms/Toggle/Toggle.native.js]: Than the area outside the toggle will also be tappable. For e.g, in your case, the toggle will be tappable outside to the right if we provide hitslop right.

File: src/atoms/Toggle/Toggle.native.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            transform: [{ translateX: knobTranslationX.current }],
+          },
+        ]}
+      />
+    </StyledContainer>
+  );
+};
+
+Toggle.propTypes = {
+  size: PropTypes.oneOf(['medium']),
```
[the remark this answers is not in the document]
Comment (shubhnik) [reply src/atoms/Toggle/Toggle.native.js]: To accommodate more sizes, the design team said they might give more sizes.

**Pull request closed without merging**